server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -4337,7 +4337,7 @@ private Pair<List<TemplateJoinVO>, Integer> searchForTemplatesInternal(ListTempl
         Account caller = CallContext.current().getCallingAccount();
         Long parentTemplateId = cmd.getParentTemplateId();
 
-        boolean listAll = false;
+        boolean listAll = cmd.listAll();
```

Comment:
should we check for who is making the request (for example, admin vs other roles)

Reply:
Let me verify this. This is similar to what is happening for list ISOs. And is being used only to decide `ListProjectResourcesCriteria` in both the APIs.

Reply:
@vishesh92 checked this?